packages/app-mining/src/components/WithdrawModal.tsx

```diff
@@ -57,6 +60,15 @@ const WithdrawModal = (props: StakePoolModalProps): JSX.Element => {
     }
   }, [])
 
+  const setMax = () => {
+    if (!userStakeInfo) return
+    const yourDelegation = format(
+      userStakeInfo.shares.mul(stakePool.totalStake.div(stakePool.totalShares)),
+      {unit: null}
+    )
+    setAmount(Number(yourDelegation))
```

Comment:
- JavaScript `number` type is not safe for `Balance`, use `string` instead.
- `format` is only used for display balance, the token decimal, which is 12 for PHA, should be used here.

Reply:
Again, the withdraw argument is **NOT stake**, but **share**

Reply:
> 

The input box is designed to enter stake here for user experience, it will be transformed to share before submitting extrinsic. See https://github.com/Phala-Network/apps/blob/master/packages/app-mining/src/components/WithdrawModal.tsx#L32